source/connection-troubleshooting.txt

```diff
@@ -204,6 +204,43 @@ Consider setting ``maxConnectionLifeTime`` and
 maintains a connection to a MongoDB instance. For more information about these
 parameters, see :ref:`Connection Pool Settings <mcs-connectionpool-settings>`.
 
+.. _java-server-selection-errors:
+
+Server Selection Errors
+-----------------------
+
+MongoDB might not be able to complete a request even when some servers
+are available, causing the driver to return a server selection error.
+The following list describes some situations in which you
+might encounter a server selection error:
+
+- You use tag sets in a read preference specification when no server
+  with those tags exists.
+- You write to a replica set when the primary node is unavailable.
+
+This error is of type ``MongoTimeoutException``. The following code
+shows a sample of the exception that occurs if you attempt to send a
+request to a replica set without a primary:
+
+.. code-block:: none
+
+   com.mongodb.MongoTimeoutException: Timed out after 30000 ms while
```

Comment:
The actual message looks a little different:

```
        com.mongodb.MongoTimeoutException:
        Timed out while waiting for a server that matches ReadPreferenceServerSelector{readPreference=primary}.
        Client view of cluster state is
        {type=REPLICA_SET,
         servers=[
          {address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {java.net.ConnectException: Connection refused}},
          {address=localhost:27018, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {java.net.ConnectException: Connection refused}},
          {address=localhost:27019, type=REPLICA_SET_SECONDARY, roundTripTime=15.0 ms, state=CONNECTED}
         ]
```

Can you update?

Reply:
Might also want to point out that the exception can be useful for troubleshooting connection issues.